service/history/execution/mutable_state_decision_task_manager_test.go

```diff
@@ -73,3 +82,365 @@ func TestReplicateDecisionTaskCompletedEvent(t *testing.T) {
+						DecisionScheduleID: 0, // pending decisions
+						DecisionAttempt:    0,
+					},
+					taskGenerator: NewMockMutableStateTaskGenerator(gomock.NewController(t)),
+					timeSource:    clock.NewMockedTimeSource(),
+				}
+			},
+			assertions: func(t *testing.T, err error, observedLogs *observer.ObservedLogs) {
+				require.NoError(t, err)
+			},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			m := &mutableStateDecisionTaskManagerImpl{msb: test.newMsb(t)}
+			core, observedLogs := observer.New(zap.DebugLevel)
+			m.msb.logger = loggerimpl.NewLogger(zap.New(core))
+			if test.expectations != nil {
+				test.expectations(m)
+			}
+			err := m.ReplicateTransientDecisionTaskScheduled()
+			test.assertions(t, err, observedLogs)
+		})
+	}
+}
+
+func TestCreateTransientDecisionEvents(t *testing.T) {
+	m := &mutableStateDecisionTaskManagerImpl{
+		msb: &mutableStateBuilder{executionInfo: &persistence.WorkflowExecutionInfo{TaskList: "some-task-list"}},
+	}
+	decision := &DecisionInfo{
+		ScheduleID:                 0,
+		StartedID:                  1,
+		RequestID:                  "some-requestID",
+		DecisionTimeout:            100,
+		Attempt:                    1,
+		ScheduledTimestamp:         10,
+		StartedTimestamp:           10,
+		OriginalScheduledTimestamp: 10,
+	}
+	scheduledEvent, startedEvent := m.CreateTransientDecisionEvents(decision, "identity")
+	require.NotNil(t, scheduledEvent)
+	assert.Equal(t, decision.ScheduleID, scheduledEvent.ID)
+	assert.Equal(t, &decision.ScheduledTimestamp, scheduledEvent.Timestamp)
+	assert.Equal(t, m.msb.executionInfo.TaskList, scheduledEvent.DecisionTaskScheduledEventAttributes.TaskList.Name)
+	assert.Equal(t, &decision.DecisionTimeout, scheduledEvent.DecisionTaskScheduledEventAttributes.StartToCloseTimeoutSeconds)
+	assert.Equal(t, decision.Attempt, scheduledEvent.DecisionTaskScheduledEventAttributes.Attempt)
+
+	require.NotNil(t, startedEvent)
+	assert.Equal(t, decision.StartedID, startedEvent.ID)
+	assert.Equal(t, &decision.StartedTimestamp, startedEvent.Timestamp)
+	assert.Equal(t, decision.ScheduleID, startedEvent.DecisionTaskStartedEventAttributes.ScheduledEventID)
+	assert.Equal(t, decision.RequestID, startedEvent.DecisionTaskStartedEventAttributes.RequestID)
+	assert.Equal(t, "identity", startedEvent.DecisionTaskStartedEventAttributes.Identity)
+}
+
+func TestGetDecisionScheduleToStartTimeout(t *testing.T) {
+	t.Run("sticky taskList", func(t *testing.T) {
+		m := &mutableStateDecisionTaskManagerImpl{
+			msb: &mutableStateBuilder{executionInfo: &persistence.WorkflowExecutionInfo{
+				StickyTaskList:               "some-sticky-task-list",
+				StickyScheduleToStartTimeout: 100,
+			}},
+		}
+		duration := m.GetDecisionScheduleToStartTimeout()
+		assert.Equal(t, time.Duration(m.msb.executionInfo.StickyScheduleToStartTimeout)*time.Second, duration)
+	})
+}
+
+func TestHasProcessedOrPendingDecision(t *testing.T) {
+	t.Run("true", func(t *testing.T) {
```

Comment:
I would recommend to name test-case after "what is there" instead of just true/false.
I think what you have in comment is great candidate: "has pending decisions"